python/rapidsmpf/rapidsmpf/config.pyx

```diff
@@ -0,0 +1,195 @@
+
+        Returns
+        -------
+        The value associated with the key, parsed and cast to the given type,
+        or the assigned default value if the key did not exist.
+
+        Raises
+        ------
+        ValueError
+            If the type of `default_value` is not supported or does not match
+            `parser_type`.
+        """
+        if issubclass(parser_type, bool):
+            return _options_get_using_parse_string[bool_t](
+                self._handle, str.encode(key), bool(default_value)
+            )
+        elif issubclass(parser_type, int):
+            return _options_get_using_parse_string[int64_t](
+                self._handle, str.encode(key), int(default_value)
+            )
+        elif issubclass(parser_type, float):
+            return _options_get_using_parse_string[double](
+                self._handle, str.encode(key), float(default_value)
+            )
+        elif issubclass(parser_type, str):
+            return _options_get_using_parse_string[string](
+                self._handle, str.encode(key), str.encode(str(default_value))
+            ).decode('UTF-8')
+        raise ValueError(
+            f"default type ({type(default_value)}) is not supported, "
+            "please use `.get()` (not implemented yet)."
+        )
+
+
+def get_environment_variables(str key_regex = "RAPIDSMPF_(.*)"):
+    """
+    Returns a dictionary of environment variables matching a given regular expression.
+
+    This function scans the current process's environment variables and inserts those
+    whose keys match the provided regular expression. The regular expression must
+    contain exactly one capture group to extract the portion of the environment variable
+    key to use as the dictionary key.
+
+    For example, to strip the `RAPIDSMPF_` prefix, use `r"RAPIDSMPF_(.*)"` as the regex.
+    The captured group will be used as the key in the output dictionary.
+
+    Example:
+        - Environment variable: RAPIDSMPF_FOO=bar
+        - key_regex: r"RAPIDSMPF_(.*)"
+        - Resulting dictionary entry: { "FOO": "bar" }
+
+    Parameters
+    ----------
+    key_regex
+        A regular expression with a single capture group to match and extract
+        the environment variable keys.
+
+    Returns
+    -------
+    A dictionary containing all matching environment variables, with keys as
+    extracted by the capture group.
+
+    Raises
+    ------
+    ValueError
+        If `key_regex` does not contain exactly one capture group.
+
+    See Also
+    --------
+    os.environ : Dictionary of the current environment variables.
+    """
+    pattern = re.compile(key_regex)
+    if pattern.groups != 1:
+        raise ValueError(
+            "key_regex must contain exactly one capture group (e.g., 'RAPIDSMPF_(.*)')"
+        )
+    ret = {}
+    for key, value in os.environ.items():
+        match = pattern.fullmatch(key)
+        if match:
+            ret[match.group(1)] = value
+    return ret
```

Comment:
Why not reuse the C++ implementation?

Reply:
To be on the safe side, I wanted the same semantic as regular Python code:

> This mapping is captured the first time the [os](https://docs.python.org/3/library/os.html#module-os) module is imported, typically during Python startup as part of processing site.py. Changes to the environment made after this time are not reflected in [os.environ](https://docs.python.org/3/library/os.html#os.environ), except for changes made by modifying [os.environ](https://docs.python.org/3/library/os.html#os.environ) directly.

And then it isn't very much code.